crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -873,17 +871,86 @@ impl<'w> EntityWorldMut<'w> {
             }
         }
 
-        #[allow(clippy::undocumented_unsafe_blocks)] // TODO: document why this is safe
+        // SAFETY: `new_archetype_id` is a subset of the components in `old_location.archetype_id`
+        // because it is created by removing a bundle from these components.
+        Self::move_entity_from_remove::<true>(
+            entity,
+            self_location,
+            old_location.archetype_id,
+            old_location,
+            entities,
+            archetypes,
+            storages,
+            new_archetype_id,
+        );
+    }
+
+    /// Removes any components in the [`Bundle`] from the entity.
+    // TODO: BundleRemover?
```

Comment:
This TODO isn't clear to me.

Reply:
I don't know what this todo is referring to, it was already on `remove` so I left it there

Reply:
Going back through the commits looks like it's referring to a potential API to aggregate `Bundle` operations into builder APIs like a `BundleRemover`.